### ISSUE TITLE
'Errors were encountered while processing'

### ISSUE THREAD

**Issue by wilml** [mathworks/build-glibc-bz-19329-patch#4]
First off thanks for the patch!

I followed the installation instructions but installing the package gave me an error but it looks like it was installed so I'm confused on the expected behavior.

ldd before install:
```
$ ldd --version ldd
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
dpkg-query before install
```
$ dpkg-query --show libc6:amd64
libc6:amd64	2.31-0ubuntu9.9
```
From the release page I installed the following package:
```
$ sudo dpkg -i libc6_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb
[sudo] password for wil: 
(Reading database ... 485834 files and directories currently installed.)
Preparing to unpack libc6_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb ...
De-configuring libc6:i386 (2.31-0ubuntu9.9) ...
Unpacking libc6:amd64 (2.31-0ubuntu9.9.ubuntu.focal.custom) over (2.31-0ubuntu9.9) ...
dpkg: error processing package libc6:amd64 (--install):
 package libc6:amd64 2.31-0ubuntu9.9.ubuntu.focal.custom cannot be configured because libc6:i386 is at a different version (2.31-0ubuntu9.9)
dpkg: error processing package libc6:i386 (--install):
 package libc6:i386 2.31-0ubuntu9.9 cannot be configured because libc6:amd64 is at a different version (2.31-0ubuntu9.9.ubuntu.focal.custom)
Errors were encountered while processing:
 libc6:amd64
 libc6:i386
```
But then I get the errors at the bottom and nothing indicating the install was successful.
However when i run,
```
$ dpkg-query --show libc6:amd64
libc6:amd64	2.31-0ubuntu9.9.ubuntu.focal.custom
```
Which makes it appear it worked. Also Matlab hasn't crashed yet which has lead me to believe it has worked. 

However if I run the following its reporting broken packages so either the install doesn't seem to be working properly or I'm executing the incorrect deb package.
```
$ sudo apt -f install
[sudo] password for wil: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... failed.
The following packages have unmet dependencies:
 libc6 : Breaks: libc6:i386 (!= 2.31-0ubuntu9.9.ubuntu.focal.custom) but 2.31-0ubuntu9.9 is installed
 libc6:i386 : Breaks: libc6 (!= 2.31-0ubuntu9.9) but 2.31-0ubuntu9.9.ubuntu.focal.custom is installed
 libc6-dbg : Depends: libc6 (= 2.31-0ubuntu9.9) but 2.31-0ubuntu9.9.ubuntu.focal.custom is installed
 libc6-dev : Depends: libc-dev-bin (= 2.31-0ubuntu9.9.ubuntu.focal.custom)
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
E: Unable to correct dependencies

```

Any clarification to the output would be helpful. Want to make sure I have this correct before helping others install it. Thanks!

**Comment by josmartin** [comment]
@wilml -- Many thanks for the report above

I've been trying to reproduce your issue in various docker containers running ubuntu-focal. I think that in addition to `libc6` it appears you also have `libc6-i386`, `libc-dev` and `libc-dbg` packages installed on your machine. The problem is that those packages depend (and are built at the same time as) `libc6` and so need to be in-sync with it. You will note that under the [current releases](https://github.com/mathworks/build-glibc-bz-19329-patch/releases) area of this repo we have also built these dependent libs. In an attempt to reproduce what you need to do I did the following ...

```
root@3eed4dc96803:/# mkdir /tmp/install
root@3eed4dc96803:/# cd /tmp/install/

root@3eed4dc96803:/tmp/install# wget -q https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/libc6_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb
root@3eed4dc96803:/tmp/install# wget -q https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/libc6-i386_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb
root@3eed4dc96803:/tmp/install# wget -q https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/libc6-dev_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb
root@3eed4dc96803:/tmp/install# wget -q https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/libc6-dbg_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb
root@3eed4dc96803:/tmp/install# wget -q https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/libc-dev-bin_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb

root@3eed4dc96803:/tmp/install# dpkg -i *.deb
(Reading database ... 9220 files and directories currently installed.)
----- SNIP -----
Setting up libc-dev-bin (2.31-0ubuntu9.9.ubuntu.focal.custom) ...
Setting up libc6-dbg:amd64 (2.31-0ubuntu9.9.ubuntu.focal.custom) ...
Setting up libc6-dev:amd64 (2.31-0ubuntu9.9.ubuntu.focal.custom) ...
Setting up libc6-i386 (2.31-0ubuntu9.9.ubuntu.focal.custom) ...
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
root@3eed4dc96803:/tmp/install#
```

Basically you are going to need to make sure you pick-up the needed set of libs for your machine. If you don't want to work out specifically which set of libs you need you can download the [all packages tarball](https://github.com/mathworks/build-glibc-bz-19329-patch/releases/download/ubuntu-focal/all-packages.tar.gz) - untar and install ALL. This tarball represents all the related packages that ubuntu thinks it needs to build alongside lib6.

Hope that helps you make progress!

**Comment by wilml** [comment]
I tried installing everything from all-packages and initial it was failing for the same reason, did a 'sudo apt --fix-broken install' and then tried again and it seemed to have worked. Thanks for the help!
